nautobot/extras/models/approvals.py

```diff
@@ -138,6 +141,48 @@ def __str__(self):
         """Stringify instance."""
         return f"{self.approval_workflow.name}: {self.object_under_review} ({self.current_state})"
 
+    @property
+    def current_stage(self):
+        """
+        Return the current stage of the workflow.
+        1. Return the stage that is denied if there is a stage that has been denied.
+        2. Return the first stage that is pending, if there is no stage that has been approved or denied.
+        3. Return the stage that is pending that is immediately after the last stage that has been approved.
+        4. Return None, if all stages are approved.
+        """
+        # Check if there is a stage that is denied
+        denied_stage = self.approval_workflow_stage_instances.filter(state=ApprovalWorkflowStateChoices.DENIED)
+        if denied_stage.exists():
+            return denied_stage.first()
+        # Get the pending stage with the lowest weight
+        pending_stages = self.approval_workflow_stage_instances.filter(
+            state=ApprovalWorkflowStateChoices.PENDING
+        ).order_by("approval_workflow_stage__weight")
+        if pending_stages.exists():
+            return pending_stages.first()
+
+        # No pending or denied stages at this point, so return None because all stages are approved
+        return None
+
+    def save(self, *args, **kwargs):
+        """
+        Override the save() method to
+        1. Update its status to denied, if one of the approval workflow stage instances is denied.
+        2. Update its status to approved, if all of the approval workflow stage instances are approved.
+        Args:
+            *args: positional arguments
+            **kwargs: keyword arguments
+        """
+        # Check if there is one or more denied response for this stage instance
+        denied_stages = self.approval_workflow_stage_instances.filter(state=ApprovalWorkflowStateChoices.DENIED)
+        if denied_stages.exists():
+            self.current_state = ApprovalWorkflowStateChoices.DENIED
+        # Check if all stages are approved
+        approved_stages = self.approval_workflow_stage_instances.filter(state=ApprovalWorkflowStateChoices.APPROVED)
+        if approved_stages.count() == self.approval_workflow.approval_workflow_stages.count():
+            self.current_state = ApprovalWorkflowStateChoices.APPROVED
```

Comment:
Feels wrong to me for this to be in the `save()` on `ApprovalWorkflowInstance` since just saving the workflow instance wouldn't normally result in a state change. I'd think that saving an `ApprovalWorkflowStageInstance` should potentially update the corresponding `ApprovalWorkflowInstance` `current_state` at that time? (I see now that you're accomplishing that by having that model's `save` call `self.approval_workflow_instance.save()` but I wonder if this logic should just be moved over there in the first place?)

(Aside, `current_stage` versus `current_state` feels like it's going to be a fun source of typo-related bugs...)